### ISSUE TITLE
dockerPushImage doesn't push image

### ISSUE THREAD

**Issue by asarkar** [bmuschko/gradle-docker-plugin#952]
### Expected Behavior
dockerPushImage should push the image built by the current project.

### Current Behavior
```
2020-07-21T13:13:45.244-0700 [DEBUG] [org.gradle.internal.execution.steps.SkipUpToDateStep] Determining if task ':dockerPushImage' is up-to-date
2020-07-21T13:13:45.244-0700 [INFO] [org.gradle.internal.execution.steps.SkipUpToDateStep] Task ':dockerPushImage' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
2020-07-21T13:13:45.244-0700 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter] Executing actions for task ':dockerPushImage'.
2020-07-21T13:13:45.244-0700 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Execute start for :dockerPushImage' started
2020-07-21T13:13:45.244-0700 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Completing Build operation 'Execute start for :dockerPushImage'
2020-07-21T13:13:45.244-0700 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Execute start for :dockerPushImage' completed
```
It's clear from the logs and the insanely fast timings that the task doesn't push the image. A manual push outputs something like the following which is absent in the logs.
```
The push refers to repository [my.company/my-project]
55ea3d7c31d8: Pushed 
d4aa72ad3fbd: Pushed 
18f4b6d754b8: Pushing [==========>                                        ]  7.666MB/35.33MB
```

### Your Environment
JDK 11, Spring Boot 2.3.1.RELEASE, Kotlin 1.3.72, plugin version 6.6.0

build.gradle.kts
```
docker {
	registryCredentials {
		username.set(Docker.USERNAME)
		password.set(Docker.PASSWORD)
	}
	springBootApplication {
		baseImage.set("azul/zulu-openjdk-alpine:11")
		images.set(setOf("my.company/${project.name}:${project.version}"))
		mainClassName.set("AppKt")
	}
}
``

**Comment by bmuschko** [comment]
As mentioned in the user guide of this plugin, the internal functionality doesn't use the Docker command line tool. It uses the Docker daemon. The behavior of the Docker daemon is different and you can't really expect the same terminal functionality.

For debugging your issue:
1. Delete the image from the remote registry.
2. Only run the push task from your Gradle build.
3. Ensure that it really doesn't push. If you pushed before, it's possible that it's really fast as the image didn't change. Check the timestamp for the Docker image in the remote registry.
4. If you can't find an issue, potentially register a callback to inspect the response from the call (see user guide).

We are pushing as part of the functional test suite so it should work in general.

**Comment by asarkar** [comment]
@bmuschko The reason I don’t think I need to execute the steps you suggested above is because I’ve run the plugin push task and the manual docker push one after another, and the behavior is completely different. The manual docker push does exactly what it’s supposed to, and indicates so through console logs. The plugin push immediately completes having done nothing, as also indicated in the logs I’ve shared.
What can cause the plugin push to not work? The project uses a SNAPSHOT version, could that be an issue?

**Comment by asarkar** [comment]
Upon registering callbacks:
```
tasks.withType<com.bmuschko.gradle.docker.tasks.image.DockerBuildImage> {
	onError { println(this) }
	onComplete { println("Docker build is successful") }
}

tasks.withType<com.bmuschko.gradle.docker.tasks.image.DockerPushImage> {
	onError { println(this) }
	onNext { println("Docker push response: ${this}") }
	onComplete { println("Docker push is successful") }
}
```

```
Successfully built 9a96c88bc024
Successfully tagged artifactory.prod.cre.az-eastus2.hosting.nuautoco.com/sandbox-docker/kt-poc:latest
Created image with ID '9a96c88bc024'.
Docker build is successful

> Task :dockerPushImage
Docker push is successful
```

So, push did absolutely nothing.

**Comment by ghostler** [comment]
Field images is empty in DockerPushImage task. Fix:
`
tasks.withType<com.bmuschko.gradle.docker.tasks.image.DockerPushImage>() {
    images.set(imgs)
}
`

**Comment by destanyinside** [comment]
> Field images is empty in DockerPushImage task. Fix:
> `tasks.withType<com.bmuschko.gradle.docker.tasks.image.DockerPushImage>() { images.set(imgs) }`

images.set(setOf(imgs)

**Comment by asarkar** [comment]
@ghostler That was it! It is a bug that push doesn't use the images specified under `springBootApplication`.

**Comment by bmuschko** [comment]
Would you mind trying out one of the previous plugin versions (e.g. 6.5.0 or 6.4.0) to verify that is a bug? I won't be get around to it until next week.

**Comment by asarkar** [comment]
@bmuschko 6.5.0 doesn't not exhibit this bug, it works as expected.

**Comment by patricklucas** [comment]
I just ran into this as well. I just upgraded from 6.5.0 to 6.6.0, and the push doesn't happen at all—the build just stops after `dockerBuildImage` completes.

Downgraded to 6.5.0 and pushing works again.

**Comment by asarkar** [comment]
@bmuschko wrote:
> We are pushing as part of the functional test suite so it should work in general.

I didn’t look into the tests, but it seems there isn’t one that verifies build and push at the same time.

**Comment by bmuschko** [comment]
Fix has been released with 6.6.1.
